### ISSUE TITLE
nRF52840 Watchdog Timer not working (ignores valid reloads)

### ISSUE THREAD

**Issue by rene-becker-setec** [renode/renode#361]
We are trying to run some FW that utilizes the nRF52840 HW watchdog. Even though the FW is running fine on an nRF52840 Dev Kit, when running in Renode we are seeing periodic resets due to the watchdog. This is using Renode V1.13

The behaviour can easily be replicated by using Zephyr's Task Watchdog sample compiled for the nRF52DK board. The following screenshot shows what happens if I modify the Zephyr sample App to never 'get stuck':
![behaviour_with_simple_sample_app](https://user-images.githubusercontent.com/77767676/180695359-769a44bd-9acb-4976-a968-5e9b3184a584.png)

The following is showing the modification to the Zephyr sample (sample located at `zephyr/samples/subsys/task_wdt/`, using Zephyr v3.1.0):

![image](https://user-images.githubusercontent.com/77767676/180695569-bcf09d69-7c6c-4976-9452-156ea10dc134.png)

Going through the Watchdog source code I see the following problems:
- There is a function to determine whether the Watchdog counter should be reloaded
  
  ![image](https://user-images.githubusercontent.com/77767676/180696181-f887498f-b4eb-49fb-9b7a-f4cd7689bfbf.png)

  however, this function is never called upon.
  I believe this function should be called whenever a (successful) write to one of the RR registers occurs, i.e. here:
  
  ![image](https://user-images.githubusercontent.com/77767676/180696651-715480ee-9658-4902-8216-4df09c2df3d1.png)

- The logic of the Reload function is not correct. I had to modify the logic as follows to get it to work
  
  ![image](https://user-images.githubusercontent.com/77767676/180697557-7f002e81-2b73-43e2-9067-a009fbd636ae.png)

  for copy & paste: 

  ```C#
       private void Reload()
        {
            if(readyToReset)
            {
                // readyToReset flag is true, which means that we already timeouted, ignore reload
                this.Log(LogLevel.Warning, "Trying to reload after timeout, ignored");
                return;
            }

            var doReload = false;
            for(var i = 0; i < NumberOfRegisters; i++)
            {
                doReload |= (requestRegisterEnabled[i] && !requestRegisterStatus[i].Value);
            }

            if(doReload)
            {
                for(var i = 0; i < NumberOfRegisters; i++)
                {
                    requestRegisterStatus[i].Value = requestRegisterEnabled[i];
                }
                Value = Limit + 1;
                this.NoisyLog("Counter reloaded");
            }
        }
  ```

With the above 2 modifications I get the simulation to behave exactly as the HW using the stock Zephyr task watchdog sample app:
![image](https://user-images.githubusercontent.com/77767676/180697270-887075b5-0405-4c14-b4f5-3f1e127fa5f1.png)

**Comment by rene-becker-setec** [comment]
Awesome. Thanks

**Comment by PiotrZierhoffer** [comment]
@rene-becker-setec you will find the fix on master with ec3efdf

**Comment by rene-becker-setec** [comment]
Great! Thanks!
